time/src/test/java/net/jqwik/api/time/MonthTests.java

```diff
@@ -0,0 +1,204 @@
+		}
+
+	}
+
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			MonthArbitrary months = Dates.months();
+			Month value = shrinkToMinimal(months, random);
+			assertThat(value).isEqualTo(Month.JANUARY);
+		}
+
+		@Property
+		void shrinksToSmallestFailingValue(@ForAll Random random) {
+			MonthArbitrary months = Dates.months();
+			TestingFalsifier<Month> falsifier = month -> month.compareTo(Month.MARCH) < 0;
+			Month value = shrinkToMinimal(months, random, falsifier);
+			assertThat(value).isEqualTo(Month.MARCH);
+		}
+
+	}
+
+	@Group
+	class ExhaustiveGeneration {
+
+		@Property(tries = 5)
+		void containsAllValues() {
+			Optional<ExhaustiveGenerator<Month>> optionalGenerator = Dates.months().exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<Month> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(12); // Cannot know the number of filtered elements in advance
+			assertThat(generator)
+					.containsExactly(Month.JANUARY, Month.FEBRUARY, Month.MARCH, Month.APRIL, Month.MAY, Month.JUNE, Month.JULY, Month.AUGUST, Month.SEPTEMBER, Month.OCTOBER, Month.NOVEMBER, Month.DECEMBER);
+		}
+
+		@Property(tries = 5)
+		void between() {
+			Optional<ExhaustiveGenerator<Month>> optionalGenerator = Dates.months().between(Month.MARCH, Month.JULY).exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<Month> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(12); // Cannot know the number of filtered elements in advance
+			assertThat(generator)
+					.containsExactly(Month.MARCH, Month.APRIL, Month.MAY, Month.JUNE, Month.JULY);
+		}
+
+		@Property(tries = 5)
+		void only() {
+			Optional<ExhaustiveGenerator<Month>> optionalGenerator = Dates.months()
+																		  .only(Month.JANUARY, Month.MARCH, Month.APRIL, Month.DECEMBER)
+																		  .exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<Month> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(12); // Cannot know the number of filtered elements in advance
+			assertThat(generator)
+					.containsExactly(Month.JANUARY, Month.MARCH, Month.APRIL, Month.DECEMBER);
+		}
+
+	}
+
+	@Group
+	class EdgeCasesTests {
+
+		@Property(tries = 5)
+		void all() {
+
+			MonthArbitrary months = Dates.months();
+			Set<Month> edgeCases = collectEdgeCases(months.edgeCases());
+			assertThat(edgeCases).hasSize(2);
+			assertThat(edgeCases).contains(Month.JANUARY, Month.DECEMBER);
+
+		}
+
+		@Property(tries = 5)
+		void between() {
+
+			MonthArbitrary months = Dates.months().between(Month.MARCH, Month.AUGUST);
+			Set<Month> edgeCases = collectEdgeCases(months.edgeCases());
+			assertThat(edgeCases).hasSize(2);
+			assertThat(edgeCases).contains(Month.MARCH, Month.AUGUST);
+
+		}
+
+	}
+
+	public static Arbitrary<Month[]> generateMonths() {
```

Comment:
Maybe I dont undestand it, but if it's supposed to generate an Array of Month, I'd do it with `monthArbitrary.array(Month[].class).ofSize...`

Reply:
The arbitrary generates arrays with different months and lengths

Reply:
So array() is the right thing I guess.

Reply:
This is my suggestion:

	public static Arbitrary<Month[]> generateMonths() {
		Arbitrary<Month> monthArbitrary = Arbitraries.of(Month.class);
		return monthArbitrary.array(Month[].class).ofMinSize(1);
	}

Better even, use a `Set<Month>` but then you don't need the whole method anymore becaus you can just say ` @ForAll @Size(min=1) Set<Month>` at point of usage.